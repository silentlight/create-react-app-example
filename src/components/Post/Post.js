import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

/**
 * Post component
 */
class Post extends PureComponent {
  render() {
    const { post } = this.props;

    return (
      <div>
        <h3>{post.get('title')}</h3>
      </div>
    );
  }
}

Post.propTypes = {
  /** Callback to be run when form submits */
  post: PropTypes.object.isRequired,
};

export default Post