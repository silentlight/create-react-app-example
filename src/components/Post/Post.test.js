import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Post from './Post';

const post = fromJS({
  title: "Test post",
});

describe('Post', () => {

  const component = shallow(<Post post={post}/>);

  it('renders and matches the snapshot', () => {
    const tree = renderer.create(<Post post={post} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the title', () => {
    expect(component.find('h3').text()).toEqual(post.get('title'));
  });

});