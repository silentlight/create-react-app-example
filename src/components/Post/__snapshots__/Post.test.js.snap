// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders and matches the snapshot 1`] = `
<div>
  <h3>
    Test post
  </h3>
</div>
`;
